extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -107,6 +109,9 @@ MigrationStateBuildItem build(BuildProducer<FeatureBuildItem> featureProducer,
 
         Collection<String> applicationMigrations = applicationMigrationsToDs.values().stream().collect(HashSet::new,
                 AbstractCollection::addAll, HashSet::addAll);
+        for (String applicationMigration : applicationMigrations) {
+            hotDeploymentProducer.produce(new HotDeploymentWatchedFileBuildItem(applicationMigration));
```

Comment:
Since 2b163bac8f52c8ed4e53add42af606966b2b4f08 it seems that we can use a glob pattern in `HotDeploymentWatchedFileBuildItem`. 

I haven't tested yet, but does that work for new files too?

Reply:
> Since [2b163ba](https://github.com/quarkusio/quarkus/commit/2b163bac8f52c8ed4e53add42af606966b2b4f08) it seems that we can use a glob pattern in `HotDeploymentWatchedFileBuildItem`.
> 
> I haven't tested yet, but does that work for new files too?

No, unfortunately it doesn't quite work for a few reasons. I reported a separate issue for that: #25338.

Reply:
@gastaldi Please also note that this PR doesn't mark the issue as fixed, since that was what the issue was about: Allow adding new files while the application is running. But if #25338 can get resolved, and then also #25337, then finally #25256 can be resolved.